src/metabase/util/schema.clj

```diff
@@ -51,6 +54,17 @@
     (recur (s/named schema api-error-message) api-error-message)
     (assoc schema :api-error-message api-error-message)))
 
+(defn with-error-message-malli
+  "Return `schema` with an additional `{:error/fn}` that will be used to explain the error if a parameter fails
+  validation."
+  [schema api-error-message]
+  {:pre [(or (vector? schema)
+             ((complement coll?) schema))]}
+  ;; TODO we're using :error/fn instead of error/message here because we wants
+  ;; malli.error/humanize to works with `deferred-tru`.
+  ;; I haven't really checked if it really works the way we want it to be though.
+  (mu/update-properties schema assoc :error/fn (fn [_ _] api-error-message)))
```

Comment:
Why not just put :error/fn (or any key) into the m.schema properties where you define them? the p.schema version of this function is necessary because that's a hack we added but since we can build erorrs / descriptions / generators / and any other arbitrary data into a m.schema I don't understand what this buys us.

Reply:
I went back and forth on this, but I agree with you, we should avoid indirection.

Let put all error/fn directly in the schema.

Reply:
made the change in https://github.com/metabase/metabase/pull/27471/commits/df50f271db1db50b6fcdcaa12ba414d4c9b26ecf